cmd/ingress-operator/start.go

```diff
@@ -118,12 +124,68 @@ func start() error {
 		os.Exit(1)
 	}
 
+	// Set up the channels for the watcher and operator.
+	stop := make(chan struct{})
+	signals := signals.SetupSignalHandler()
+
+	// Set up and start the file watcher.
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Error(err, "failed to create watcher")
+		os.Exit(1)
+	}
+	defer watcher.Close()
+	if err := watcher.Add(defaultTrustedCABundle); err != nil {
+		log.Error(err, "failed to add file to watcher", "filename", defaultTrustedCABundle)
+		os.Exit(1)
+	}
+	log.Info("watching file", "filename", defaultTrustedCABundle)
+	go func() {
+		orig, err := ioutil.ReadFile(defaultTrustedCABundle)
+		if err != nil {
+			log.Error(err, "failed to read watched file", "filename", defaultTrustedCABundle)
+			close(stop)
+			return
+		}
```

Comment:
This `ReadFile` and error check might as well go outside the goroutine (then it can use `os.Exit(1)`).

Reply:
ack